### PR TITLE
Revert data migration to backfill notifications

### PR DESCRIPTION
**Pull request by prabhanshuguptagit**
**Story card:** [ch3585](https://app.clubhouse.io/simpledotorg/story/3585/backfill-notifications-from-communications)

## Because

The data migration to backfill `Notification` goes through all `Communication`s sequentially which is very slow. The data migration would be expected to take a few days. 

## This addresses

The migration is not required for `Notification` to be released. This drops the backfill data migration to unblock master. We should do the backfill at some point in a way that works for prod – IHCI has ~10m communications. 

Maintenance story:
https://app.clubhouse.io/simpledotorg/story/3585/backfill-notifications-from-communications

### CODE REVIEW

**Comment by harimohanraj89** [comment]
@prabhanshuguptagit do we need to rollback the migration on your local machine? I expected to see the `data_schema.rb` file reverted as well.

**Comment by prabhanshuguptagit** [comment]
@harimohanraj89 The `data_schema` was never committed to master, so we're good. No need to revert it.

File: db/schema.rb [hunk 1]
```diff
@@ -141,6 +141,7 @@
     t.string "detailable_type"
     t.bigint "detailable_id"
     t.uuid "notification_id"
+    t.index ["appointment_id"], name: "index_communications_on_appointment_id"
```
Comment (prabhanshuguptagit) [inline db/schema.rb:144]: This is a previously uncommitted schema change.

File: db/schema.rb [hunk 2]
```diff
@@ -130,7 +130,7 @@
   end
 
   create_table "communications", id: :uuid, default: -> { "gen_random_uuid()" }, force: :cascade do |t|
-    t.uuid "appointment_id", null: false
+    t.uuid "appointment_id"
```
Comment (harimohanraj89) [inline db/schema.rb:133]: As discussed offline, this is because a previous migration didn't check in its update to `schema.rb`

**Review by harimohanraj89 — APPROVED** [review]
:shipit:

**Pull request merged**